datafusion/core/src/execution/options.rs

```diff
@@ -115,7 +128,57 @@ impl<'a> CsvReadOptions<'a> {
             collect_stat: false,
             file_extension: self.file_extension.to_owned(),
             target_partitions,
+            table_partition_cols: self.table_partition_cols.clone(),
+        }
+    }
+}
+
+/// Parquet read options
+#[derive(Clone)]
+pub struct ParquetReadOptions<'a> {
+    /// File extension; only files with this extension are selected for data input.
+    /// Defaults to ".parquet".
+    pub file_extension: &'a str,
+    /// Partition Columns
+    pub table_partition_cols: Vec<String>,
+    /// Should DataFusion parquet reader using the predicate to prune data, following execution::context::SessionConfig
```

Comment:
```suggestion
    /// Should DataFusion parquet reader using the predicate to prune data, 
    /// overridden by value on execution::context::SessionConfig
```

Maybe after this PR is merged, we can remove the option on `SessionConfig`